source/common/config/delta_subscription_impl.h

```diff
@@ -48,51 +41,42 @@ class DeltaSubscriptionImpl : public Subscription,
     request_.mutable_node()->MergeFrom(local_info_.node());
   }
 
-  // Enqueues and attempts to send a discovery request, (un)subscribing to resources missing from /
-  // added to the passed 'resources' argument, relative to resource_versions_.
-  void buildAndQueueDiscoveryRequest(const std::vector<std::string>& resources) {
-    ResourceNameDiff diff;
-    std::set_difference(resources.begin(), resources.end(), resource_names_.begin(),
-                        resource_names_.end(), std::inserter(diff.added_, diff.added_.begin()));
-    std::set_difference(resource_names_.begin(), resource_names_.end(), resources.begin(),
-                        resources.end(), std::inserter(diff.removed_, diff.removed_.begin()));
-
-    for (const auto& added : diff.added_) {
-      setResourceWaitingForServer(added);
+  void updateResources(std::vector<std::string> resource_names) override {
+    std::vector<std::string> cur_added;
+    std::vector<std::string> cur_removed;
+
+    // set_difference expects sorted collections. (resource_names_ is always sorted; it's std::set).
+    std::sort(resource_names.begin(), resource_names.end());
+
+    std::set_difference(resource_names.begin(), resource_names.end(), resource_names_.begin(),
```

Comment:
This is kinda hard to read since `resource_names` and `resources_names_` is hard to visually differentiate, maybe rename one of them?

Reply:
Very true! Done.